app/src/pages/playground/playgroundUtils.ts

```diff
@@ -1043,3 +1058,98 @@ export function normalizeMessageAttributeValue(
     ? content
     : JSON.stringify(content, null, 2);
 }
+
+export function areRequiredInvocationParametersConfigured(
+  configuredInvocationParameters: InvocationParameterInput[],
+  supportedInvocationParameters: InvocationParameter[]
+) {
+  return supportedInvocationParameters
+    .filter((param) => param.required)
+    .every((param) =>
+      configuredInvocationParameters.some((ip) =>
+        areInvocationParamsEqual(ip, param)
+      )
+    );
+}
+
+/**
+ * Extracts the default value for the invocation parameter definition
+ * And the key name that should be used in the invocation parameter input if we need to make a new one
+ *
+ * This logic is necessary because the default value is mapped to different key name based on its type
+ * within the InvocationParameterInput queries in the playground
+ */
+const getInvocationParamDefaultValue = (
+  param: InvocationParameter
+): {
+  invocationInputField: string | undefined;
+  defaultValue: unknown;
+} => {
+  let defaultValueKeyName: string | undefined;
+  Object.entries(param).find(([key, value]) => {
+    if (key.endsWith("DefaultValue") && value != null) {
+      defaultValueKeyName = key;
+      return true;
+    }
+    return false;
+  });
+  return {
+    invocationInputField: param.invocationInputField,
+    defaultValue:
+      defaultValueKeyName != null
+        ? param[defaultValueKeyName as keyof InvocationParameter]
+        : undefined,
+  };
+};
+
+/**
+ * Merges the current invocation parameters with the default values for the supported invocation parameters,
+ * only adding values for invocation parameters that don't already have a value
+ */
+export function mergeInvocationParametersWithDefaults(
+  invocationParameters: InvocationParameterInput[],
+  supportedInvocationParameters: InvocationParameter[]
+) {
+  // Convert the current invocation parameters to a map for quick lookup
+  const currentInvocationParametersMap = new Map(
+    invocationParameters.map((param) => [
+      param.canonicalName || param.invocationName,
+      param,
+    ])
+  );
+  supportedInvocationParameters.forEach((param) => {
+    const paramKeyName = param.canonicalName || param.invocationName;
+    // Extract the default value for the invocation parameter definition
+    // And the key name that should be used in the invocation parameter input if we need to make a new one
+    const { invocationInputField, defaultValue } =
```

Comment:
this invocationInputField seems unnecessary, it's just pulling directly from the param so it's basically just a pass through on that function, we can just use the param below

Reply:
all the checks / usages of it down below can just be replaced with param.invocationInputField, just to be clear the param.invocationInputField is the name of the field that points to the value correct? and this is not just value because it's like stringValue booleanValue etc.? If that's the case it seems cumbersome but maybe htere is not other option with unions and stuff in the api or the other tradeoffs are not worth it, been a while since we did the original

Reply:
Yup good callout. Simplified the usage of this field. This fn went through multiple refactors before pushing and I could not clean it up well enough when I was "done"